hexes/hexes.py

```diff
@@ -51,6 +54,26 @@ def __repr__(self):
         ).format(s=self)
 
     @property
+    def text(self):
+        return self._text
+
+    @text.setter
+    def text(self, val):
+        dirty = val != self._text
```

Comment:
Er, isn't that desirable? If it hasn't changed, not dirty—not if it hasn't been assigned to, not dirty.